src/Interpreters/Access/InterpreterGrantQuery.cpp

```diff
@@ -117,7 +117,7 @@ namespace
     void checkGranteeIsAllowed(const ContextAccess & current_user_access, const UUID & grantee_id, const IAccessEntity & grantee)
     {
         auto current_user = current_user_access.getUser();
-        if (current_user && !current_user->grantees.match(grantee_id))
+        if (!current_user || !current_user->grantees.match(grantee_id))
```

Comment:
If `current_user` is null that means this is a context without a user, i.e. this is the global context and the global context must not be restricted.

Reply:
Initially this change was made by @Enmk, but it looked correct to me. When can we execute `GRANT` query in global context? Why it's not a logical error? (for example, when we accidentally passed some `context` instead of some `local_context` to some function and this `context` was global one)